init.lua

```diff
@@ -844,7 +844,12 @@ textadept.editing.autocompleters.lsp = function()
 			label = symbol.filterText or symbol.label
 			snippets[label] = symbol.insertText
 		end
-		-- TODO: some labels can have spaces and need proper handling.
+		
+		-- If a label contains spaces, use '|' as the separator. TODO: This is not ideal. A label might contain '|' too.
```

Comment:
Yes, you're right. I still need to sit and think on this. When I'm testing, I often type `buffer:auto_c_show(0, 'foo bar baz')` into the command entry to popup an autocompletion list, and this inherently relies on the default value of ' '. I should probably rewire my thinking and use `auto_c_show(0,table.concat({'foo','bar','baz'},string.byte(auto_c_separator)))`, but that's a lot of typing :)

Still, thanks for bringing this up and I will put it on my TODO list.

Reply:
The solution I currently settled in is to patch lsp module for it to use textredux for filename autocompletion functionality. (which was the use case requiring proper handling of spaces). 

Personally I see this workaround as a good enough fix. Even if not ideal.

I didn’t close the PR, it automatically got closed when I deleted my fork btw. Reopening now. Do what you want with it.